### PR TITLE
Room Test Case and Async Method Implementation on API Side is done

### PR DESCRIPTION
**Pull request by sathish199308**
Room Test Case and Async Method Implementation on API Side is done

### CODE REVIEW

File: BookingSystem.DAL/Repository/RoomRepository.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public RoomRepository(DataContext dataContext)
             _dataContext = dataContext;
         }
 
-        public List<RoomsVM> GetRoomDetails()
+        public async Task<List<RoomsVM>> GetRoomDetails()
         {
             var VenueDetails = (from vm in _dataContext.Rooms
```
Comment (manic1990) [inline BookingSystem.DAL/Repository/RoomRepository.cs:24]: Check IsDeleted condition also.

File: BookingSystemAPI/Controllers/RoomController.cs [hunk 2]
```diff
@@ -59,22 +61,26 @@ public async Task<ActionResult> Get()
         [Route("saverooms")]
         public async Task<ActionResult> Post(RoomsVM rooms)
         {
+            _logger.LogInformation(nameof(RoomController) + nameof(Post), started);
             var response = new ResponseModel();
             try
             {
-                var result = await _roomService.SaveRoomDetails(rooms);
```
Comment (manic1990) [inline BookingSystemAPI/Controllers/RoomController.cs:65]: Add ConfigureAwait(false).

File: BookingSystem.DAL/Repository/RoomRepository.cs (hunk 1 above)
Comment (sathish199308) [reply BookingSystem.DAL/Repository/RoomRepository.cs:24]: IsDeleted Condition is checked

File: BookingSystemAPI/Controllers/RoomController.cs (hunk 2 above)
Comment (sathish199308) [reply BookingSystemAPI/Controllers/RoomController.cs:65]: Configure Await (false) is added

**Pull request merged**